src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1022,7 +1022,11 @@ void DataBox<tmpl::list<Tags...>>::unlock_box_after_mutate() {
 template <typename... Tags>
 template <typename ImmutableItemTag>
 SPECTRE_ALWAYS_INLINE bool DataBox<tmpl::list<Tags...>>::reset_compute_item() {
-  // reference items do not need to be reset, but it
+  // reference items do not need to be reset, but it still needs to be handled
+  // here. Partly just so we get an error if we are calling this function
+  // incorrectly and because we need to call this function during resets
+  // to get dependency tracking correctly. When compute tags depend on reference
```

Comment:
...tracking correct when compute tags...

Reply:
I squashed this change into the fixup. Waiting for @wthrowe before squashing the fixups in